target/ppc/translate/storage-ctrl-impl.c.inc

```diff
@@ -21,6 +21,198 @@
+}
+
+static bool do_slbia(DisasContext *ctx, arg_X_ih *a)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+
+#if defined(TARGET_PPC64)
+    gen_helper_slbia(cpu_env, tcg_constant_i32(a->ih));
+#else
+    qemu_build_not_reached();
+#endif
+#endif
+    return true;
+}
+
+static bool do_slbiag(DisasContext *ctx, arg_X_rs_l *a)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+
+#if defined(TARGET_PPC64)
+    gen_helper_SLBIAG(cpu_env, cpu_gpr[a->rs], tcg_constant_i32(a->l));
+#else
+    qemu_build_not_reached();
+#endif
+#endif
+    return true;
+}
+
+static bool do_slbmte(DisasContext *ctx, arg_X_sb *a)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+
+#if defined(TARGET_PPC64)
+    gen_helper_store_slb(cpu_env, cpu_gpr[a->rb], cpu_gpr[a->rs]);
+#else
+    qemu_build_not_reached();
+#endif
+#endif
+    return true;
+}
+
+static bool do_slbmfev(DisasContext *ctx, arg_X_sb *a)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+
+#if defined(TARGET_PPC64)
+    gen_helper_load_slb_vsid(cpu_gpr[a->rs], cpu_env,
+                             cpu_gpr[a->rb]);
+#else
+    qemu_build_not_reached();
+#endif
+#endif
+    return true;
+}
+
+static bool do_slbmfee(DisasContext *ctx, arg_X_sb *a)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+
+#if defined(TARGET_PPC64)
+    gen_helper_load_slb_esid(cpu_gpr[a->rs],cpu_env, cpu_gpr[a->rb]);
+#else
+    qemu_build_not_reached();
+#endif
+#endif
+    return true;
+}
+
+static bool do_slbfee(DisasContext *ctx, arg_X_tb1 *a)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    gen_inval_exception(ctx, POWERPC_EXCP_PRIV_REG);
+#else
+
+#if defined(TARGET_PPC64)
+    TCGLabel *l1, *l2;
+    if (unlikely(ctx->pr)) {
+        gen_inval_exception(ctx, POWERPC_EXCP_PRIV_REG);
+        return true;
+    }
+    gen_helper_find_slb_vsid(cpu_gpr[a->rt], cpu_env,
+                             cpu_gpr[a->rb]);
+    l1 = gen_new_label();
+    l2 = gen_new_label();
+    tcg_gen_trunc_tl_i32(cpu_crf[0], cpu_so);
+    tcg_gen_brcondi_tl(TCG_COND_EQ, cpu_gpr[a->rt], -1, l1);
+    tcg_gen_ori_i32(cpu_crf[0], cpu_crf[0], CRF_EQ);
+    tcg_gen_br(l2);
+    gen_set_label(l1);
+    tcg_gen_movi_tl(cpu_gpr[a->rt], 0);
+    gen_set_label(l2);
+#else
+    qemu_build_not_reached();    
+#endif
+#endif
+    return true;
+}
+
+static bool do_slbsync(DisasContext *ctx)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+
+#if defined(TARGET_PPC64)
+    gen_check_tlb_flush(ctx, true);
```

Comment:
CHK_SV is missing

Reply:
I couldn't find the fix in 1a0d54c. Would the fix be for the comment above, perhaps?

Reply:
It looks like CHK_SV is still missing in this case (`do_slbsync`).

Reply:
Latest version is still missing a `CHK_SV`, before `gen_check_tlb_flush`, in `trans_SLBSYNC`.

Reply:
Fixed in b1e0c02